src/containers/course-section/resource-item/ResourceItem.tsx

```diff
@@ -59,109 +54,102 @@ const ResourceItem: FC<ResourceItemProps> = ({
   editResource,
   isView = false
 }) => {
-  const onDeleteResource = () => {
+  const handleDeleteResource = () => {
     deleteResource?.(resource)
   }
 
-  const onEditResource = () => {
+  const handleEditResource = () => {
     editResource?.(resource)
   }
 
   const { t } = useTranslation()
 
-  const { resourceAvailability, isCooperation } =
+  const { resourceAvailability: allResourcesAvailability, isCooperation } =
     useResourceAvailabilityContext()
 
-  const setResourceIcon = (): ReactElement | undefined => {
+  const renderResourceIcon = () => {
     if (resource.resourceType === ResourcesTypes.Lessons) {
       return resourcesData.lessons.icon
     } else if (resource.resourceType === ResourcesTypes.Quizzes) {
       return resourcesData.quizzes.icon
     }
   }
```

Comment:
I guess it can be improved.
What do you think ?

Reply:
I found this solution rather good for this use case. I also considered following solution:

```jsx
let icon;

if (resource.resourceType === ResourcesTypes.Lessons) {
     icon = resourcesData.lessons.icon
}

if (resource.resourceType === ResourcesTypes.Quizzes) {
     icon = resourcesData.quizzes.icon
}
```

Then we can just pass icon to child component.

Maybe I don't see better solution here. Can you please recommend a better one?

Reply:
> I found this solution rather good for this use case. I also considered following solution:
> 
> ```js
> let icon;
> 
> if (resource.resourceType === ResourcesTypes.Lessons) {
>      icon = resourcesData.lessons.icon
> }
> 
> if (resource.resourceType === ResourcesTypes.Quizzes) {
>      icon = resourcesData.quizzes.icon
> }
> ```
> 
> Then we can just pass icon to child component.
> 
> Maybe I don't see better solution here. Can you please recommend a better one?

As I see `switch case` also fits and became more readable and changeable.
For example. If we have another resource type. We can simple add new case.
What I mean

```jsx
const renderResourceIcon = () => {
  switch (resource.resourceType) {
    case ResourcesTypes.Lessons:
      return resourcesData.lessons.icon
    case ResourcesTypes.Quizzes:
      return resourcesData.quizzes.icon
    default:
      return null
  }
}
```

Reply:
Personally I don't think so. I am pretty sure that list of resources will not change in the nearest future. Maximum that can be added are attachments, but it will be very easy to add them with current implementation.

Maybe I should just add `return null` as a default value and it will be useful improvement

Reply:
> just add `return null` as a default value

What do you mean add `return null` ? 
In my case, if you use switch case it's necessary to return a default value. if just to use this approach

Reply:
I mean like following:

```jsx
const renderResourceIcon = () => {
    if (resource.resourceType === ResourcesTypes.Lessons) {
      return resourcesData.lessons.icon
    } else if (resource.resourceType === ResourcesTypes.Quizzes) {
      return resourcesData.quizzes.icon
    }
    
    return null
  }
```

Reply:
> I mean like following:
> 
> ```js
> const renderResourceIcon = () => {
>     if (resource.resourceType === ResourcesTypes.Lessons) {
>       return resourcesData.lessons.icon
>     } else if (resource.resourceType === ResourcesTypes.Quizzes) {
>       return resourcesData.quizzes.icon
>     }
>     
>     return null
>   }
> ```

but it's difficult to read. Are u agree ?

Reply:
I partially agree, because for instance we can remove `else if` and refactor as following:

```jsx
const renderResourceIcon = () => {
    if (resource.resourceType === ResourcesTypes.Lessons) {
      return resourcesData.lessons.icon
    } 

   if (resource.resourceType === ResourcesTypes.Quizzes) {
      return resourcesData.quizzes.icon
    }
    
    return null
  }
```

What are your thoughts about it?

Reply:
> I partially agree, because for instance we can remove `else if` and refactor as following:
> 
> ```js
> const renderResourceIcon = () => {
>     if (resource.resourceType === ResourcesTypes.Lessons) {
>       return resourcesData.lessons.icon
>     } 
> 
>    if (resource.resourceType === ResourcesTypes.Quizzes) {
>       return resourcesData.quizzes.icon
>     }
>     
>     return null
>   }
> ```
> 
> What are your thoughts about it?

using switch case. It's like one line (case...) but this approach it's almost 3 lines.
This approach also make sense. 
We'll discuss it

Reply:
I used your proposal for now after my own brainstorm after our meeting. We should rewrite cooperation activities first and then we can change it if needed in the future

Reply:
> I used your proposal for now after my own brainstorm after our meeting. We should rewrite cooperation activities first and then we can change it if needed in the future
> 
> Now it looks like this:
> 
> ```js
> const renderResourceIcon = () => {
>     const { Lessons, Quizzes } = ResourcesTypes
> 
>     switch (resource.resourceType) {
>       case Lessons:
>         return resourcesData.lessons.icon
>       case Quizzes:
>         return resourcesData.quizzes.icon
>       default:
>         return null
>     }
>   }
> ```

Good looking!
using destructuring for `ResourcesTypes` nice case